### PR TITLE
Allow building docker with early access JDKs

### PR DESCRIPTION
**Pull request by wendigo**
Adds some basic JDK 22-ea support

### CODE REVIEW

**Comment by wendigo** [comment]
Docker tests on 22-ea have passed

File: core/docker/build.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ Builds the Trino Docker image
 -a       Build the specified comma-separated architectures, defaults to amd64,arm64,ppc64le
 -r       Build the specified Trino release version, downloads all required artifacts
 -j       Build the Trino release with specified Temurin JDK release
+-e       Build the Trino release with the early access JDK release
```
Comment (nineinchnick) [inline core/docker/build.sh:14]: Can we extract this from the version (22-ea) to avoid adding yet another flag?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -121,6 +121,8 @@ jobs:
           platforms: arm64,ppc64le
       - name: Build and Test Docker Image
         run: core/docker/build.sh
+      - name: Build and Test Docker Image with JDK 22-ea
+        run: core/docker/build.sh -j 22 -e
```
Comment (nineinchnick) [inline .github/workflows/ci.yml:125]: The container tests we run are very basic smoke tests, but building the image is expensive. Is it really worth running in every PR? Even with all the recent improvements in test run times, we can still easily create a backlog in the CI.

File: core/docker/build.sh (hunk 1 above)
Comment (wendigo) [reply core/docker/build.sh:14]: We potentially could

File: .github/workflows/ci.yml (hunk 2 above)
Comment (wendigo) [reply .github/workflows/ci.yml:125]: The purpose of this test is to build a docker image and check whether it starts

File: core/docker/build.sh (hunk 1 above)
Comment (wendigo) [reply core/docker/build.sh]: Done

**Review by electrum — APPROVED** [review]

**Pull request merged**